WISE-Community/WISE-Client
# chore(Login): Use recaptcha V3

## Changes

If Recaptcha keys are set in application.properties, when a user logs in, verify with Recaptcha V3 (invisible).

## Test

- Test with https://github.com/WISE-Community/WISE-API/pull/228

When Recaptcha is enabled
- This message is displayed near the bottom of the login page "This site is protected by reCAPTCHA and the Google Privacy Policy and Terms of Service apply."
- When Recaptcha verification fails, make sure it shows the Recaptcha error message "Recaptcha failed. Please reload the page and try again!"
  - You can do this by changing this line
  - https://github.com/WISE-Community/WISE-Client/blob/ebbd750ffba0585d440484f13828ba71749905f6/src/app/services/user.service.ts#L89
  - To this
`httpParams = httpParams.set('recaptchaResponse', 'abc'); `
  - You can also test the request by using Postman
    - Set the request to POST to http://localhost:81/api/j_acegi_security_check
    - Set the Body form-data to

| KEY | VALUE |
| - | - |
| username | your-valid-username |
| password | your-correct-password |
| recaptchaResponse | any-random-incorrect-value |
- When the user types in a wrong username or password, make sure it shows the message "Username and password not recognized. Please try again."
- When a user types in a correct username and password, they should be logged in and directed to the user home page

When Recaptcha is disabled
- This message is not displayed "This site is protected by reCAPTCHA and the Google Privacy Policy and Terms of Service apply."
- When the user types in a wrong username or password, make sure it shows the message "Username and password not recognized. Please try again."
- When a user types in a correct username and password, they should be logged in and directed to the user home page

Closes #1312
